### PR TITLE
Revert "fix On iOS, dialog titles are announced twice (#19826)"

### PR DESCRIPTION
**Pull request by gaaclarke**
This reverts commit 48a6fde39f9fe8bf7ad06f0f8cf690186d930b73.

## Description

See https://github.com/flutter/flutter/issues/67345#issuecomment-705851312 for details.

## Related Issues

https://github.com/flutter/flutter/issues/67345
https://github.com/flutter/flutter/issues/44001

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [x] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
LGTM

**Review by liyuqian — APPROVED** [review]

**Comment by gaaclarke** [comment]
@chunhtai can you double check my latest patch (https://github.com/flutter/engine/pull/21714/commits/8f2ccd24a0c6902231469c6d97eb8ad15ce3682a)  I had to change a few tests.  I think one of the tests was broken, it was asserting that the argument to the notification was "route" but it never set that string anywhere.   I'm guessing it was relying on previous tests to set that state?

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge_test.mm [hunk 1]
```diff
@@ -394,9 +393,9 @@ - (void)testAnnouncesRouteChangesWhenNoNamesRoute {
 
   // Notification should focus first focusable node, which is node1.
   XCTAssertEqual([accessibility_notifications count], 1ul);
-  SemanticsObject* focusObject = accessibility_notifications[0][@"argument"];
-  XCTAssertEqual([focusObject uid], 2);
-  XCTAssertEqualObjects([focusObject accessibilityLabel], @"node2");
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge_test.mm:399]: This seems wrong in retrospect, right?  Node2 wasn't a route but was getting the route change notificaiton call?

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge_test.mm [hunk 2]
```diff
@@ -328,7 +328,7 @@ - (void)testAnnouncesRouteChanges {
   bridge->UpdateSemantics(/*nodes=*/nodes, /*actions=*/actions);
 
   XCTAssertEqual([accessibility_notifications count], 1ul);
-  XCTAssertEqualObjects(accessibility_notifications[0][@"argument"], @"route");
+  XCTAssertEqualObjects(accessibility_notifications[0][@"argument"], @"node3");
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge_test.mm:331]: This was just a bad revert I think.

Comment (chunhtai) [reply shell/platform/darwin/ios/framework/Source/accessibility_bridge_test.mm:331]: yes, this looks good

**Pull request merged**